### PR TITLE
Combine /records and /country_seroprev_summaries endpoint into a single one

### PR DESCRIPTION
**Pull request by atatmaja**
## Briefly describe the feature or bug that this PR addresses.

See title, this was done to speed up serotracker.com (by preventing unnecessary computation)

Note: We've temporarily moved the combined endpoint into /dashboard_records. This was done under request by @rahularoradfs because research doesn't currently have BW to update their code to pull records (I think there's an urgent deliverable on their end).

## Please link the Airtable ticket associated with this PR.

https://airtable.com/appT8tnLbGJV6v81V/tbli2lWQHAqBa6ZcI/viwWvxb27XGuziw4c/recoOARzx3RuUghug?blocks=bip9X54YrLFYz3ycP

## Describe the steps you took to test the feature/bugfix introduced by this PR.

Updated unit tests, ran application locally and tested against local web deployment and tested using postman.

### CODE REVIEW

File: app/namespaces/data_provider/data_provider_controller.py [hunk 1]
```diff
@@ -67,6 +67,85 @@ def post(self):
...
+        prioritize_estimates_mode = data.get('prioritize_estimates_mode', 'dashboard')
+        include_disputed_regions = data.get('include_disputed_regions', False)
+        include_subgeography_estimates = data.get('include_subgeography_estimates', False)
+        unity_aligned_only = data.get('unity_aligned_only', False)
+        include_records_without_latlngs = data.get('include_records_without_latlngs', False)
+        calculate_country_seroprev_summaries = data.get('calculate_country_seroprev_summaries', True)
+
+        sampling_start_date, sampling_end_date = convert_start_end_dates(data, use_sampling_date=True)
+        publication_start_date, publication_end_date = convert_start_end_dates(data, use_sampling_date=False)
+        include_in_srma = data.get('include_in_srma', False)
```
Comment (simonarocco) [inline app/namespaces/data_provider/data_provider_controller.py:109]: move line 109 to be close to lines 97 to 105 bc they are doing similar things

File: app/namespaces/data_provider/data_provider_controller.py [hunk 2]
```diff
@@ -67,6 +67,85 @@ def post(self):
...
+
+        columns = columns_requested
```
Comment (simonarocco) [inline app/namespaces/data_provider/data_provider_controller.py:111]: just add a comment here please about why we do this on line 111

File: app/namespaces/data_provider/data_provider_controller.py [hunk 3]
```diff
@@ -67,6 +67,85 @@ def post(self):
...
+        # If we intend to calculate country summaries we will need certain columns
+        if calculate_country_seroprev_summaries and columns:
+            COUNTRY_SEROPREV_SUMMARY_COLS = ['country', 'country_iso3', 'denominator_value', 'serum_pos_prevalence',
```
Comment (simonarocco) [inline app/namespaces/data_provider/data_provider_controller.py:114]: set this const var to lowercase plz

**Review by simonarocco — APPROVED** [review]
lgtm, small nits, dkm

File: app/namespaces/data_provider/data_provider_controller.py (hunk 3 above)
Comment (atatmaja) [reply app/namespaces/data_provider/data_provider_controller.py:114]: why

**Pull request merged**